purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -1128,13 +1158,36 @@ internal class PurchasesOrchestrator(
         isPersonalizedPrice: Boolean?,
         purchaseCallback: PurchaseCallback,
     ) {
+        trackPurchaseStarted(purchasingData.productId, purchasingData.productType)
+        val startTime = dateProvider.now
+
+        val callbackWithDiagnostics = object : PurchaseCallback {
```

Comment:
We could potentially do a small optimization that if `diagnosticsTrackerIfEnabled` is null, we can just fallback to the `purchaseCallback` to avoid the extra callback. Not a huge deal though.

Reply:
Yeah, I see. It makes sense. I'll do it!

Related: can you think of any way to avoid the duplicated code (this `callbackWithDiagnostics` and the above `listenerWithDiagnostics` are the same). I could think of some solutions but I think they would be overcomplicated to what they achieve, tbh 🤔

Reply:
Right! I was thinking of more complicated solutions for some reason... 🤦
Well, I'm glad I asked 😅 
Thank you!